components/prism-core.js

```diff
@@ -280,9 +280,15 @@ var _ = _self.Prism = {
 					lookbehindLength = 0,
 					alias = pattern.alias;
 
+				if (greedy && !pattern.pattern.global) {
+					// Without the global flag, lastIndex won't work
+					pattern.pattern = RegExp(pattern.pattern.source, pattern.pattern.flags + "g");
```

Comment:
@Golmote Yes you are right it is not supported well at all. Good catch. Your solution is perfect and we should add it.